internal/otlpenv/otlpenv.go

```diff
@@ -0,0 +1,56 @@
+// Package otlpenv exports getters to read OpenTelemetry protocol configuration options
+// based on the official spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options
+package otlpenv
+
+import (
+	"os"
+	"strings"
+)
+
+// getWithDefault returns def if no env in keys is set, or the first env from keys that is
+// set.
+func getWithDefault(def string, keys ...string) string {
+	for _, k := range keys {
+		if v, set := os.LookupEnv(k); set {
+			return v
+		}
+	}
+	return def
+}
+
+// This is a custom default that's also not quite compliant but hopefully close enough (we
+// use 127.0.0.1 instead of localhost, since there's a linter rule banning localhost).
+const defaultGRPCCollectorEndpoint = "http://127.0.0.1:4317"
```

Comment:
FWIW we can add exceptions to that linter, https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/dev/check/no-localhost-guard.sh?L15-21

Reply:
Yep! But there _is_ a rationale behind that, so I think it best to just leave this as is for now